lib/factorial.rb

```diff
@@ -1,6 +1,15 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) because number of iterations is dependent on the size of the input number
+# Space complexity: O(1) because only one int value (factorial) is ever stored
 def factorial(number)
-  raise NotImplementedError
+  if number
+    factorial = 1
+    while number > 0
+      factorial *= number
+      number -= 1
+    end
+    return factorial
+  else
+    raise ArgumentError, "Input must be an integer"
```

Comment:
I'd recommend having error handling at the top rather than the bottom so you break early. Something such as:
```
if !number 
  raise
else 
  do factorial
end
```

Otherwise good job on this assignment!